src/services/user-group/hooks/index.js

```diff
@@ -45,33 +41,30 @@ const addWholeClassToCourse = (hook) => {
  * @param hook - contains and request body
  */
 const deleteWholeClassFromCourse = (hook) => {
-	let requestBody = hook.data;
-	let courseId = hook.id;
-	return CourseModel.findById(courseId).exec().then(course => {
+	const requestBody = hook.data;
+	const courseId = hook.id;
+	if (requestBody.classIds === undefined) {
+		return hook;
+	}
+	return CourseModel.findById(courseId).exec().then((course) => {
 		if (!course) return hook;
 
-		let removedClasses = _.differenceBy(course.classIds, requestBody.classIds, (v) => JSON.stringify(v));
+		const removedClasses = _.differenceBy(course.classIds, requestBody.classIds, v => JSON.stringify(v));
 		if (removedClasses.length < 1) return hook;
-		return Promise.all(removedClasses.map(classId => {
+		return Promise.all(removedClasses.map((classId) => {
 			return ClassModel.findById(classId).exec().then(c => (c || []).userIds);
-		})).then(studentIds => {
+		})).then(async (studentIds) => {
 			// flatten deep arrays and remove duplicates
 			studentIds = _.uniqWith(_.flattenDeep(studentIds), (e1, e2) => JSON.stringify(e1) === JSON.stringify(e2));
 
-			// remove all students of classes from course, if they are in course
-			return Promise.all(studentIds.map(s => {
-				if (!_.some(course.userIds, u => JSON.stringify(u) === JSON.stringify(s))) {
-					return CourseModel.update({ _id: course._id }, { $pull: { userIds: s } }).exec();
-				} else {
-					return {};
-				}
-			})).then(result => {
-
-				// also remove all students from request body for not reading them in after hook
-				requestBody.userIds = _.differenceBy(requestBody.userIds, studentIds, (v) => JSON.stringify(v));
-				hook.data = requestBody;
-				return hook;
-			});
+			// remove class students from course DB and from hook body to not patch them back
+			await CourseModel.update(
+				{ _id: course._id },
+				{ $pull: { userIds: { $in: studentIds } } },
+				{ multi: true },
+			).exec();
+			hook.data.userIds = hook.data.userIds.filter(value => !studentIds.toString().includes(value));
```

Comment:
stringifying the array might work, but IMO it would be safer to use something like `!studentIds.some(id => id.toString() === value)`